lib/renderer/chart_painter.dart

```diff
@@ -69,31 +68,43 @@ class ChartPainter extends BaseChartPainter {
 
   @override
   void initChartRenderer() {
-    if (fixedLength == null) {
-      if (datas == null || datas.isEmpty) {
+    if (fixedLength == null && datas != null) {
```

Comment:
Origin is:
```
 if (fixedLength == null) {      if (datas == null || datas.isEmpty) {        fixedLength = 2;      } else {        var t = datas[0];        fixedLength =            NumberUtil.getMaxDecimalLength(t.open, t.close, t.high, t.low);      }    }
```
Error.In your code, when `datas` is null, `fixedLength ` will be null.

Reply:
i changed `int? fixedLength` to `int fixedLength` and assign 2 as default